p2p/src/sync/index.rs

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+use crate::{
+    error::P2pError,
+    sync::{
+        mock_consensus::{BlockHeader, BlockId},
+        queue::{ImportQueue, ImportQueueState, QueuedData},
+    },
+};
+use logging::log;
+use std::collections::HashMap;
+
+// TODO: use LRU cache for the import queue
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum PeerIndexState {
+    /// Block has been accepted to peer's block index
+    Accepted,
+
+    /// Ancestor missing, block has been queued
+    Queued,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct InnerPeerIndex {
+    pub id: BlockId,
+    pub prev_id: Option<BlockId>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct PeerIndex {
```

Comment:
Can you please document these structs and functions? It's really difficult to understand the context and purpose from the outside... so one has to dig deep to see what's going on.

Is this supposed to be the "PeerState"?

Reply:
I think this piece of code is the one that'll evolve the most during the next few weeks so I wouldn't concentrate on this too much. This is the intermediary index that holds the peer's chain but at the moment it's not optimized and doesn't even serve much of a purpose so it'll be reworked quite a bit.